samples/wifi/shell/src/wifi_cred_shell.c

```diff
@@ -0,0 +1,229 @@
+ */
+
+#include <zephyr/kernel.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/init.h>
+
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_event.h>
+
+#include <net/wifi_credentials.h>
+#include <net/wifi_mgmt_ext.h>
+
+#define MACSTR "%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx"
+
+static void print_network_info_helper(void *cb_arg, const char *ssid, size_t ssid_len)
+{
+	int ret = 0;
+	struct wifi_credentials_personal creds = { 0 };
+
+	ret = wifi_credentials_get_by_ssid_personal_struct(ssid, ssid_len, &creds);
+	if (ret) {
+		printk("an error occurred when trying to load credentials for network \"%.*s\""
+			". err: %d", ssid_len, ssid, ret);
+		return;
+	}
+	printk("  network ssid: \"%.*s\", ssid_len: %d, type: %s",
+		ssid_len, ssid, ssid_len,
+		wifi_security_txt(creds.header.type));
+
+
+	if (creds.header.type == WIFI_SECURITY_TYPE_PSK ||
+	    creds.header.type == WIFI_SECURITY_TYPE_PSK_SHA256 ||
+	    creds.header.type == WIFI_SECURITY_TYPE_SAE) {
+		printk(", password: \"%.*s\", password_len: %d",
+		       creds.password_len, creds.password, creds.password_len);
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_BSSID) {
+		printk(", bssid: " MACSTR,
+		       creds.header.bssid[0], creds.header.bssid[1],
+		       creds.header.bssid[2], creds.header.bssid[3],
+		       creds.header.bssid[4], creds.header.bssid[5]);
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_2_4GHz) {
+		printk(", band: 2.4GHz");
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_5GHz) {
+		printk(", band: 5GHz");
+	}
+
+	if (creds.header.flags & WIFI_CREDENTIALS_FLAG_FAVORITE) {
+		printk(", favorite");
+	}
+	printk("\n");
+}
+
+static enum wifi_security_type parse_sec_type(const char *s)
+{
+	if (strcmp("OPEN", s) == 0) {
+		return WIFI_SECURITY_TYPE_NONE;
+	}
+	if (strcmp("WPA2-PSK", s) == 0) {
+		return WIFI_SECURITY_TYPE_PSK;
+	}
+	if (strcmp("WPA2-PSK-SHA256", s) == 0) {
+		return WIFI_SECURITY_TYPE_PSK_SHA256;
+	}
+	if (strcmp("WPA3-SAE", s) == 0) {
+		return WIFI_SECURITY_TYPE_SAE;
+	}
+	return WIFI_SECURITY_TYPE_UNKNOWN;
+}
+
+static enum wifi_frequency_bands parse_band(const char *s)
+{
+	if (strcmp("2.4GHz", s) == 0) {
+		return WIFI_FREQ_BAND_2_4_GHZ;
+	}
+	if (strcmp("5GHz", s) == 0) {
+		return WIFI_FREQ_BAND_5_GHZ;
+	}
+	if (strcmp("6GHz", s) == 0) {
+		return WIFI_FREQ_BAND_6_GHZ;
+	}
+	return WIFI_FREQ_BAND_UNKNOWN;
+}
+
+static int cmd_auto_connect(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	int rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, iface, NULL, 0);
+
+	if (rc) {
+		printk("an error occurred when trying to auto-connect to a network. err: %d", rc);
+	}
+	return 0;
+}
+
+static int cmd_add_network(const struct shell *sh, size_t argc, char *argv[])
+{
+	int ret;
+
+	if (argc < 3) {
+		goto help;
+	}
+
+	struct wifi_credentials_personal creds = {
```

Comment:
```suggestion
	if (strnlen(argv[1], WIFI_SSID_MAX_LEN + 1) > WIFI_SSID_MAX_LEN) {
		printk("SSID too long");
		goto help;
	}

	struct wifi_credentials_personal creds = {
```
We should check the length of the SSID input

Reply:
done